### ISSUE TITLE
[Bug]: Using `int` type in knn function, core dumped.

### ISSUE THREAD

**Issue by chrysanthemum-boy** [infiniflow/infinity#774]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

7257da23

### Actual behavior

```
[12:02:59.510] [8724] [critical] Not implemented@src/function/table/knn_scan_data.cpp:74
Unrecoverable error issued, stop the serverShutdown infinity server ...
[12:02:59.549] [8724] [critical] Error: SEGMENT FAULTS
[12:02:59.550] [8724] [critical] 0, ./build/src/infinity(___interceptor_backtrace+0x75) [0x64121efb80a5]
[12:02:59.550] [8724] [critical] 1, ./build/src/infinity(+0x183a42e) [0x64122028d42e]
[12:02:59.550] [8724] [critical] 2, ./build/src/infinity(+0x5fd528) [0x64121f050528]
[12:02:59.550] [8724] [critical] 3, /lib/x86_64-linux-gnu/libc.so.6(+0x42990) [0x7ac12d042990]
[12:02:59.550] [8724] [critical] 4, ./build/src/infinity(+0x9e0ee2) [0x64121f433ee2]
[12:02:59.550] [8724] [critical] 5, ./build/src/infinity(+0x9e0ea5) [0x64121f433ea5]
[12:02:59.551] [8724] [critical] 6, ./build/src/infinity(+0x9bf2f5) [0x64121f4122f5]
[12:02:59.551] [8724] [critical] 7, ./build/src/infinity(+0xe1d179) [0x64121f870179]
[12:02:59.551] [8724] [critical] 8, ./build/src/infinity(+0x9a282e) [0x64121f3f582e]
[12:02:59.551] [8724] [critical] 9, ./build/src/infinity(+0x82b3be) [0x64121f27e3be]
[12:02:59.551] [8724] [critical] 10, ./build/src/infinity(+0x820685) [0x64121f273685]
[12:02:59.551] [8724] [critical] 11, ./build/src/infinity(+0x86a14e) [0x64121f2bd14e]
[12:02:59.551] [8724] [critical] 12, ./build/src/infinity(+0x25175d3) [0x641220f6a5d3]
[12:02:59.551] [8724] [critical] 13, ./build/src/infinity(+0x24d3b7c) [0x641220f26b7c]
[12:02:59.551] [8724] [critical] 14, ./build/src/infinity(+0x24ca6c1) [0x641220f1d6c1]
[12:02:59.551] [8724] [critical] 15, ./build/src/infinity(+0x25099da) [0x641220f5c9da]
[12:02:59.551] [8724] [critical] 16, ./build/src/infinity(+0x25093fe) [0x641220f5c3fe]
[12:02:59.551] [8724] [critical] 17, ./build/src/infinity(+0x250929d) [0x641220f5c29d]
[12:02:59.551] [8724] [critical] 18, ./build/src/infinity(+0x2509272) [0x641220f5c272]
[12:02:59.551] [8724] [critical] 19, ./build/src/infinity(+0x2509235) [0x641220f5c235]
[12:02:59.551] [8724] [critical] 20, ./build/src/infinity(+0x2508fe9) [0x641220f5bfe9]
[12:02:59.551] [8724] [critical] 21, ./build/src/infinity(+0x3db6943) [0x641222809943]
[12:02:59.551] [8724] [critical] 22, ./build/src/infinity(+0x5b8e9b) [0x64121f00be9b]
[12:02:59.551] [8724] [critical] 23, /lib/x86_64-linux-gnu/libc.so.6(+0x97b5a) [0x7ac12d097b5a]
[12:02:59.552] [8724] [critical] 24, /lib/x86_64-linux-gnu/libc.so.6(+0x1285fc) [0x7ac12d1285fc]
[12:03:00.469] [8660] [info] DELTA Checkpoint Txn txn_id: 1241, begin_ts: 2480, max_commit_ts 2478
[12:03:00.469] [8660] [info] SAVING DELTA CATALOG
[12:03:00.469] [8660] [info] Save delta catalog commit ts:2478, checkpoint max commit ts:2478.
terminate called without an active exception
Aborted (core dumped)
```

### Steps to reproduce

```Markdown
Function test case can be found in python/test/test_knn.py `test_various_embedding_data_type`
please use `int` embedding_data_type to reproduce the bugs.
```

**Comment by Ognimalf** [comment]
The current Python SDK does not support type conversion between different numerical types, which will make python side throw the exceptions. I think it's necessary to support the feature.

**Comment by JinHai-CN** [comment]
Issue the error message: "integer embedding element type hasn't been supported."

**Comment by Ognimalf** [comment]
Fixed by #847 .

**Comment by chrysanthemum-boy** [comment]
It looks fixed.
